packages/protocol/test/governance/downtime_slasher_intervals.ts

```diff
@@ -0,0 +1,731 @@
+            ),
+            'not down'
+          )
+        })
+      })
+      describe('when the validator was down', () => {
+        it('success with validator index change', async () => {
+          await slasher.setEpochSigner(epoch - 1, 1, validatorList[0])
+          const slotArrays = await makeBlockInfoSlashable(startBlock, [1, validatorIndexInEpoch])
+          await slasher.slash(
+            slotArrays.startSlots,
+            slotArrays.endSlots,
+            [1, validatorIndexInEpoch],
+            0,
+            [],
+            [],
+            [],
+            [],
+            [],
+            []
+          )
+          const balance = await mockLockedGold.accountTotalLockedGold(validatorList[0])
+          assert.equal(balance.toNumber(), 40000)
+        })
+        it('success without validator index change', async () => {
+          await slasher.setEpochSigner(epoch - 1, validatorIndexInEpoch, validatorList[0])
+          const slotArrays = await makeBlockInfoSlashable(startBlock, [
+            validatorIndexInEpoch,
+            validatorIndexInEpoch,
+          ])
+
+          await slasher.slash(
+            slotArrays.startSlots,
+            slotArrays.endSlots,
+            [validatorIndexInEpoch, validatorIndexInEpoch],
+            0,
+            [],
+            [],
+            [],
+            [],
+            [],
+            []
+          )
+          const balance = await mockLockedGold.accountTotalLockedGold(validatorList[0])
+          assert.equal(balance.toNumber(), 40000)
+        })
+        it("fails if the indeces don't match the same validator", async () => {
+          await slasher.setEpochSigner(epoch - 1, 1, validatorList[0])
+          await slasher.setEpochSigner(epoch, 1, validatorList[1])
+          const slotArrays = await makeBlockInfoSlashable(startBlock, [1, validatorIndexInEpoch])
+          await assertRevert(
+            slasher.slash(
+              slotArrays.startSlots,
+              slotArrays.endSlots,
+              [1, 1],
+              0,
+              [],
+              [],
+              [],
+              [],
+              [],
+              []
+            ),
+            'indices do not point to the same validator'
+          )
+        })
+      })
+      describe('when succeds', () => {
+        beforeEach(async () => {
+          await slasher.setEpochSigner(epoch - 1, validatorIndexInEpoch, validatorList[0])
+          const slotArrays = await makeBlockInfoSlashable(startBlock, [
+            validatorIndexInEpoch,
+            validatorIndexInEpoch,
+          ])
+          await slasher.slash(
+            slotArrays.startSlots,
+            slotArrays.endSlots,
+            [validatorIndexInEpoch, validatorIndexInEpoch],
+            0,
+            [],
+            [],
+            [],
+            [],
+            [],
+            []
+          )
+        })
+        it('cannot be slashed twice if it shares at least a block', async () => {
```

Comment:
Who shares a block with whom?

Reply:
Changed it to this:
'cannot be slashed twice if it shares at least a block with the slash of another epoch'

This is a tricky case. We will have only one slash per epoch, but it is possible to have two consecutive epochs, and a previous slash covering most of the interval.
Example: (epoch of 10 blocks, and slashable downtime of 5. Epoch 1: 1-10, Epoch 2: 11-20)
The validator was down from 9-15
A user slashes the validator from 9-13 (we save that as a slash for the epoch1)
A user (same or another, it's the same) tries to slash the validator from 11-15. The validator was down for 5 blocks, and it's in another epoch, but it was previously penalised for those "blocks down".